File: Heuristic Search (sgware)/src/edu/uno/ai/planning/h/StateHeuristic.java
package edu.uno.ai.planning.h;

import edu.uno.ai.planning.State;
import edu.uno.ai.planning.Heuristic;
import edu.uno.ai.planning.ss.StateSpaceProblem;

public abstract class StateHeuristic implements Heuristic<State> {

	public final StateSpaceProblem problem;
	
	public StateHeuristic(StateSpaceProblem problem) {
		this.problem = problem;
	}
	
	@Override
	public abstract double evaluate(State current);
}

File: UnitySokoban/Assets/Scripts/Planning/HeuristicSearchPlannerSGW/StateHeuristic.cs
using Planning;
using StateSpaceSearchProject;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace HeuristicSearchPlannerSGW
{
    public abstract class StateHeuristic : Heuristic<State> {

        public readonly StateSpaceProblem problem;

        public StateHeuristic(StateSpaceProblem problem)
        {
            this.problem = problem;
        }

        public abstract int evaluate(State current);
    }

}
